### PR TITLE
[nrf mergeup] Upmerge w17 2020

### PR DESCRIPTION
**Pull request by nvlsianpu**
Upmerge up to  https://github.com/zephyrproject-rtos/mcuboot/commit/80a76f4af4314635bef83272779ddbed6021fc32

### CODE REVIEW

**Comment by nvlsianpu** [comment]
unfortunately NCS base was #8a6e16ec79845f9c821e8ec2048193f213f47096, mea culpa, need to rebase :(

**Comment by nvlsianpu** [comment]
^^ upmerge rebased

File: zephyr/Kconfig [hunk 1]
```diff
@@ -23,6 +23,16 @@ config MCUBOOT_IMAGE_VERSION
...
 
+# HACK: NCS temphack to keep our imgtool integration working now that
+# there is no CONFIG_DT_* CMake namespace anymore. Use Zephyr
+# kconfigfunctions to thread the flash write block size through
+# Kconfig so it can be accessed from CMake. Needs a better solution.
+DT_ZEPHYR_FLASH := zephyr,flash
+DT_CHOSEN_ZEPHYR_FLASH := $(dt_chosen_path,$(DT_ZEPHYR_FLASH))
+config MCUBOOT_FLASH_WRITE_BLOCK_SIZE
+	hex
+	default $(dt_node_int_prop_int,$(DT_CHOSEN_ZEPHYR_FLASH),write-block-size)
```
Comment (nvlsianpu) [inline zephyr/Kconfig:34]: @mbolivar-nordic 
Shouldn't this use `dt_node_int_prop_hex()` as property type is hex?

Comment (nvlsianpu) [reply zephyr/Kconfig]: @mbolivar-nordic - fixed in newer commit, will squash once the fix will be approved.

Comment (nvlsianpu) [reply zephyr/Kconfig:34]: nop, should be `int` type

Comment (mbolivar-nordic) [reply zephyr/Kconfig:34]: Thanks for checking this and the fix.

**Pull request merged**